injector/injector.go

```diff
@@ -0,0 +1,312 @@
+					encoder.Flush()
+					outputXML.WriteString("<TrackingEvents>")
+					for typ, urls := range ti.events.LinearTrackingEvents {
+						ti.provider.PopulateEventMacros(creativeId, "", typ)
+						for _, url := range urls {
+							outputXML.WriteString("<Tracking event=\"")
+							outputXML.WriteString(typ)
+							outputXML.WriteString("\"><![CDATA[")
+							ti.replacer.ReplaceBytes(&outputXML, url, ti.provider)
+							outputXML.WriteString("]]></Tracking>")
+						}
+					}
+					outputXML.WriteString("</TrackingEvents>")
+				}
+				encoder.EncodeToken(tt)
+				continue
+			case "InLine", "Wrapper":
+				wrapperTagFound = false
+				inlineWrapperTagFound = true
+				encoder.Flush()
+
+				if !impressionTagFound {
+					for _, url := range ti.events.Impressions {
+						outputXML.WriteString("<Impression><![CDATA[")
+						ti.replacer.ReplaceBytes(&outputXML, url, ti.provider)
+						outputXML.WriteString("]]></Impression>")
+					}
+				}
+				impressionTagFound = false
+
+				if !errorTagFound {
+					for _, url := range ti.events.Errors {
+						outputXML.WriteString("<Error><![CDATA[")
+						ti.replacer.ReplaceBytes(&outputXML, url, ti.provider)
+						outputXML.WriteString("]]></Error>")
+
+					}
+				}
+				errorTagFound = false
+
+				encoder.EncodeToken(tt)
+			case "NonLinear":
+				encoder.Flush()
+
+				ti.provider.PopulateEventMacros(creativeId, "", "")
+				for _, url := range ti.events.NonLinearClickTracking {
+					outputXML.WriteString("<NonLinearClickTracking><![CDATA[")
+					ti.replacer.ReplaceBytes(&outputXML, url, ti.provider)
+					outputXML.WriteString("]]></NonLinearClickTracking>")
+				}
+				nonLineaerTagFound = true
+				encoder.EncodeToken(tt)
+			case "Companion":
+				companionTagFound = true
+				encoder.Flush()
+				ti.provider.PopulateEventMacros(creativeId, "", "")
+				for _, url := range ti.events.CompanionClickThrough {
+					outputXML.WriteString("<CompanionClickThrough><![CDATA[")
+					ti.replacer.ReplaceBytes(&outputXML, url, ti.provider)
+					outputXML.WriteString("]]></CompanionClickThrough>")
+				}
+				encoder.EncodeToken(tt)
+			case "Creative":
+				isCreative = false
+			case "CompanionAds":
+				if !companionTagFound && wrapperTagFound {
+					outputXML.WriteString("<Companion>")
+					for _, url := range ti.events.CompanionClickThrough {
+						outputXML.WriteString("<CompanionClickThrough><![CDATA[")
+						ti.replacer.ReplaceBytes(&outputXML, url, ti.provider)
+						outputXML.WriteString("]]></CompanionClickThrough>")
+					}
+					outputXML.WriteString("<Companion>")
+				}
+			}
+
+		case xml.CharData:
+			tt2 := strings.Trim(string(tt), trimRunes)
+			if len(tt2) != 0 {
+				encoder.Flush()
+				outputXML.WriteString("<![CDATA[")
+				outputXML.WriteString(tt2)
+				outputXML.WriteString("]]>")
+				continue
+			}
+		}
+
+		encoder.EncodeToken(t)
+	}
+
+	encoder.Flush()
+
+	if !inlineWrapperTagFound {
+		// 	// Todo log adapter.<bidder-name>.requests.badserverresponse metrics
```

Comment:
Is this TODO completed, if not, can you address it?

Reply:
Could you update this TODO comment for formatting:

i.e: `//TODO: log adapter.<bidder-name>.requests.badserverresponse metrics`